airbyte-integrations/bases/debezium/src/test/java/io/airbyte/integrations/debezium/internals/DebeziumShutdownProcedureTest.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.debezium.internals;
+
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.atomic.AtomicInteger;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+public class DebeziumShutdownProcedureTest {
+
+  @Test
+  public void test() throws InterruptedException {
+    final LinkedBlockingQueue<Integer> sourceQueue = new LinkedBlockingQueue<>(10);
+    final AtomicInteger recordsInserted = new AtomicInteger();
+    final ExecutorService executorService = Executors.newSingleThreadExecutor();
+    final DebeziumShutdownProcedure<Integer> debeziumShutdownProcedure = new DebeziumShutdownProcedure<>(sourceQueue,
+        executorService::shutdown, () -> recordsInserted.get() >= 99);
+    executorService.execute(() -> {
+      for (int i = 0; i < 100; i++) {
+        try {
+          sourceQueue.put(i);
+          recordsInserted.set(i);
+        } catch (InterruptedException e) {
+          throw new RuntimeException(e);
+        }
+      }
+    });
+
+    Thread.sleep(1000);
+    debeziumShutdownProcedure.initiateShutdownProcedure();
+
+    Assertions.assertEquals(99, recordsInserted.get());
+    Assertions.assertEquals(0, sourceQueue.size());
+    Assertions.assertEquals(100, debeziumShutdownProcedure.getRecordsRemainingAfterShutdown().size());
+
+    for (int i = 0; i < 99; i++) {
+      Assertions.assertEquals(i, debeziumShutdownProcedure.getRecordsRemainingAfterShutdown().poll());
```

Comment:
How is this happening? Is `getRecordsRemainingAfterShutdown` not being actually remaining but the total?
If we have already inserted 99 records, the source queue is empty, and the total expected by the for loop is 100... Remaining should be 1?

Also what is happening with last record, are we losing it?

Reply:
Ah its just a mistake on my end. The loop should be <100. My bad.